test.js

```diff
@@ -447,3 +450,60 @@ test('alias - require previousId and userId', t => {
     })
   })
 })
+
+test.cb('e2e', t => {
+  const id = uid(16)
+
+  const testFn = function () {
+    var axiosClient = axios.create({
+      baseURL: 'https://api.runscope.com',
+      timeout: 10 * 1000,
+      headers: { 'Authorization': 'Bearer ' + process.env.RUNSCOPE_TOKEN }
+    })
+
+    retries(axiosClient, { retries: 3 })
+
+    axiosClient.get('buckets/zfte7jmy76oz/messages?count=10')
+        .then(function (response) {
+          t.is(response.status, 200)
+
+          // TODO: fail and end the test if none of the requests match.
```

Comment:
We could set a timeout on the client, which would cause an error and result in failure.

Reply:
Not sure why the timeout on the client would help? The requests won't match if the library had a bug,api was down, etc.

Reply:
> Currently the test will just hang forever if this happens

I'm suggesting we call `.end()` after a specified amount of time. One way to do this would be to have Axios throw an error after N seconds.

Reply:
So the async stuff fixed this!